### PR TITLE
new branch: my not diamond

### PR DESCRIPTION
**Pull request by Krzysiek31415**

### CODE REVIEW

**Comment by ziobron** [comment]
You should deliver changes to diamonds branch, not master.

File: Car.hpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#pragma once
+#include "Engine.hpp"
+class Car
+{
+public:
+    Car();
+    virtual ~Car();
```
Comment (nauka-programowania-MA) [inline Car.hpp:7]: rule of 5

File: Car.hpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    
+    void turnLeft();
+    void turnRight();
+    void brake();
+    void accelerate(int speed);
+    void refuel();
+    //virtual void setEngine(Engine * engine) = 0;
+
+    //PetrolEngine* engine_;
```
Comment (nauka-programowania-MA) [inline Car.hpp:16]: remove commented code

File: ElectricCar.cpp [hunk 3]
```diff
@@ -8,9 +8,11 @@ ElectricCar::ElectricCar(ElectricEngine* engine)
 }
 
 ElectricCar::~ElectricCar()       { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::turnLeft()      { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::turnRight()     { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::brake()         { std::cout << __FUNCTION__ << std::endl; }
-void ElectricCar::accelerate(int) { std::cout << __FUNCTION__ << std::endl; }
 void ElectricCar::charge()        { std::cout << __FUNCTION__ << std::endl; }
 
+void ElectricCar::setEngine(ElectricEngine * engine)
```
Comment (nauka-programowania-MA) [inline ElectricCar.cpp:13]: maybe work on unique_ptr?

File: ElectricCar.hpp [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
+class ElectricCar : public Car
 {
 public:
     ElectricCar(ElectricEngine* engine);
-    ~ElectricCar();
-    void turnLeft();
-    void turnRight();
-    void brake();
-    void accelerate(int speed);
+    ~ElectricCar() override;
```
Comment (nauka-programowania-MA) [inline ElectricCar.hpp:8]: rule of 5

File: ElectricEngine.hpp [hunk 5]
```diff
@@ -1,11 +1,13 @@
...
+#include "Engine.hpp"
 
-class ElectricEngine
+class ElectricEngine : public Engine
 {
 public:
     ElectricEngine(int power, int batteryCapacity);
-
-    int power_;             // in HP 
+    ~ElectricEngine() override;
```
Comment (nauka-programowania-MA) [inline ElectricEngine.hpp:8]: rule of 5

File: Engine.hpp [hunk 6]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+#include "Engine.hpp"
+
+class Engine 
+{
+public:
+    Engine(int power);
+    virtual ~Engine();
```
Comment (nauka-programowania-MA) [inline Engine.hpp:8]: rule of 5

File: HybridCar.hpp [hunk 7]
```diff
@@ -1,18 +1,17 @@
...
+class HybridCar : public Car
 {
 public:
     HybridCar(PetrolEngine* petrolEng, ElectricEngine* electricEng);
-    ~HybridCar();
-    void turnLeft();
-    void turnRight();
-    void brake();
-    void accelerate(int speed);
+    ~HybridCar() override;
```
Comment (nauka-programowania-MA) [inline HybridCar.hpp:9]: rule of 5

File: PetrolCar.hpp [hunk 8]
```diff
@@ -1,16 +1,15 @@
...
 {
 public:
     PetrolCar(PetrolEngine* engine);
-    ~PetrolCar();
-    void turnLeft();
-    void turnRight();
-    void brake();
-    void accelerate(int speed);
-    void refuel();
+    ~PetrolCar() override;
```
Comment (nauka-programowania-MA) [inline PetrolCar.hpp:8]: rule of 5

File: PetrolEngine.hpp [hunk 9]
```diff
@@ -1,12 +1,15 @@
 #pragma once
+#include "Engine.hpp"
 
-class PetrolEngine
+class PetrolEngine : public Engine
 {
 public:
     PetrolEngine(int power, float capacity, int gears);
+    ~PetrolEngine() override;
```
Comment (nauka-programowania-MA) [inline PetrolEngine.hpp:8]: rule of 5

File: main.cpp [hunk 10]
```diff
@@ -5,24 +5,25 @@
...
-    toyota.accelerate(100);
-    toyota.brake();
-    toyota.charge();
-    toyota.refuel();
+    // std::cout << std::endl << "TOYOTA" << std::endl;
+    // HybridCar toyota(new PetrolEngine(80, 1400, 5), new ElectricEngine(100, 540));
+    // toyota.accelerate(100);
+    // toyota.brake();
+    // toyota.charge();
+    // toyota.refuel();
```
Comment (nauka-programowania-MA) [inline main.cpp:28]: better write some test :)

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**